runtime/lua/vim/lsp/util.lua

```diff
@@ -1361,6 +1362,45 @@ local position_sort = sort_by_key(function(v)
   return {v.start.line, v.start.character}
 end)
 
+-- Gets the zero-indexed line from the given uri.
+-- For non-file uris, we load the buffer and get the line
+-- If a loaded buffer exists, then that is used.
+-- Otherwise we get the line using libuv which is a lot faster than loading the buffer.
+--@param uri string uri of the resource to get the line from
+--@param row number zero-indexed line number
+--@return string the line at row in filename
+function M.get_line(uri, row)
+  -- load the buffer if this is not a file uri
+  -- NOTE: when does this happen?
+  if uri:sub(1, 4) ~= "file" then
+    local bufnr = vim.uri_to_bufnr(uri)
+    vim.fn.bufload(bufnr)
```

Comment:
One thing that popped up in other PRs where `bufload` was implicitly added was the question of whether it should be unloaded again after the `nvim_buf_get_lines call

Not sure which behavior makes more sense in this case. I'd tend to do an unload because it brings the fs-read and the buf-load read behavior closer together.

Reply:
I'm not sure about this one, since typically for references etc, you might get hits for the same file, but multiple lines, so we'd have to reload those buffers.
 
Although, in this case (for non-file) buffers, it does probably make sense 🤔

Reply:
Good point.  :+1:  for keeping it loaded then.